vignettes/fims-path-maturity.Rmd

```diff
@@ -0,0 +1,549 @@
+knitr::include_graphics("figures/fims-path-maturity-2.png")
+```
+
+
+## Inside the `fims::LogisticMaturity` class
+
+The `LogisticMaturity` class in the `fims` namespace defined in [logistic.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/population_dynamics/maturity/functors/logistic.hpp) under [inst/include/population_dynamics/maturity/functors](https://github.com/NOAA-FIMS/FIMS/tree/main/inst/include/population_dynamics/maturity/functors) has an `evaluate` method (i.e., function) that takes an input, *x* and returns the output from a logistic function (defined in [fims_math.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/common/fims_math.hpp) under [inst/include/common](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/common)) using the class member `median` and `slope` values.  
+
+```{Rcpp, eval = FALSE}
+namespace fims {
+
+/**
+ *  @brief LogisticMaturity class that returns the logistic function value
+ * from fims_math.
+ */
+template <typename T>
+struct LogisticMaturity : public MaturityBase<T> {
+  T median; /*!< 50% quantile of the value of the quantity of interest (x); e.g.
+               age at which 50% of the fish are mature */
+  T slope;  /*!<scalar multiplier of difference between quantity of interest
+               value (x) and median */
+
+  LogisticMaturity() : MaturityBase<T>() {}
+
+  /**
+   * @brief Method of the logistic maturity class that implements the
+   * logistic function from FIMS math.
+   *
+   * @param x  The independent variable in the logistic function (e.g., age or
+   * size at maturity).
+   */
+  virtual const T evaluate(const T& x) {
+    return fims::logistic<T>(median, slope, x);
+  }
+};
+
+}  // namespace fims
+```
+
+```{r image-path-maturity-3, echo = FALSE, message=FALSE, out.width = '100%'}
+knitr::include_graphics("figures/fims-path-maturity-3.png")
+```
+
+## Calling maturity from population
+
+The `Population` class defined in [population.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/population_dynamics/population/population.hpp) under [inst/include/population_dynamics/population](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/population_dynamics/population) is where all the biological calculations happen, producing expected values used in likelihood equations and derived values important for management (e.g. SSB). 
+
+To evaluate maturity within the `Population` class, we first need to set up a shared pointer, linking the maturity module to population.
+
+```{Rcpp, eval = FALSE}
+// file: inst/include/population_dynamics/population/population.hpp
+// maturity is a shared pointer to MaturityBase
+int maturity_id = -999; /*!< id of maturity model object*/
+std::shared_ptr<fims::MaturityBase<Type>>
+      maturity; /*!< shared pointer to maturity module */
+```
+
+The `maturity_id` was set from R using the `SetMaturity()` method from the `PopulationInterface` class: `fims$population$SetMaturity(fims$maturity$get_id())`
```

Comment:
I noticed that this fims$population$SetMaturity(fims$maturity$get_id()) is not technically what "we" do earlier in the vignette on line 44. Maybe it does the same thing though.

Reply:
fixed - this should be the same as line 44